src/NTH.Windows.Forms/NTH.Windows.Forms/NthListView.cs

```diff
@@ -1,48 +1,111 @@
-        protected override CreateParams CreateParams
-        {
-            get
-            {
-                var cp = base.CreateParams;
-                if(Environment.OSVersion.Version.Major >= 6)
-                    cp |= 0; // TODO
-                return cp;
-            }
-        }
+        private const int LVM_FIRST = 0x1000;
+        private const int LVM_SETEXTENDEDLISTVIEWSTYLE = LVM_FIRST + 54;
+        private const int LVS_EX_DOUBLEBUFFER = 0x00010000;
+        private bool _isExplorerListView;
+
+        //protected override CreateParams CreateParams
+        //{
+        //    get
+        //    {
+        //        var cp = base.CreateParams;
+        //        if(Environment.OSVersion.Version.Major >= 6)
+        //            cp |= 0; // TODO
+        //        return cp;
+        //    }
+        //}
+
         public NthListView()
         {
-            SetBuffingOptions();
+            SetStyle(ControlStyles.OptimizedDoubleBuffer | ControlStyles.AllPaintingInWmPaint, true);
+            //if (Environment.OSVersion.Version.Major < 6)
+            //    SetStyle(ControlStyles.UserPaint, true);
+            UpdateStyles();
+
+            FullRowSelect = true; // Customizable...
         }
 
-        private void SetBuffingOptions()
+        //protected override void OnPaint(PaintEventArgs e)
+        //{
+        //    if (GetStyle(ControlStyles.UserPaint))
+        //    {
+        //        Message m = new Message
+        //        {
+        //            HWnd = Handle,
+        //            Msg = WM_PRINTCLIENT,
+        //            WParam = e.Graphics.GetHdc(),
+        //            LParam = (IntPtr) PRF_CLIENT
+        //        };
+        //        DefWndProc(ref m);
+        //        e.Graphics.ReleaseHdc(m.WParam);
+        //    }
+        //    base.OnPaint(e);
+        //}
+
+        public void MakeCollapsable() // Adds expanders to the groups, should be called in form's Show-event
         {
-            SetStyle(ControlStyles.OptimizedDoubleBuffer | ControlStyles.AllPaintingInWmPaint, true);
-            // Disable default CommCtrl painting on non-Vista systems
             if (Environment.OSVersion.Version.Major < 6)
-                SetStyle(ControlStyles.UserPaint, true);
+                return;
+
+            const int lvmFirst = 0x1000;
+            const int LVM_SETGROUPINFO = (lvmFirst + 147);
+
+            foreach (ListViewGroup group in Groups)
+            {
+                var placeHolderGroup = new NthListViewGroup();
+                placeHolderGroup.CbSize = Marshal.SizeOf(placeHolderGroup);
+                placeHolderGroup.State = NthListViewGroupState.Collapsible;
+                placeHolderGroup.Mask = NthListViewGroupMask.State;
+                placeHolderGroup.GroupId = GetGroupId(group);
+
+                if (placeHolderGroup.GroupId >= 0)
+                    NativeMethods.SendMessage(Handle, LVM_SETGROUPINFO, new IntPtr(placeHolderGroup.GroupId), ref placeHolderGroup);
```

Comment:
Oh, well. Yes. The struct is allocated on the stack, so the GC shouldn't move it.
Since pinvokes cannot be generic (even if the type is constrained to a struct), we have to retain the overload :/.